src/piped/specs.clj

```diff
@@ -0,0 +1,41 @@
+(ns piped.specs
+  (:require [clojure.spec.alpha :as s]))
+
+(s/def :piped/client any?)
+(s/def :piped/queue-url string?)
+(s/def :piped/producer-parallelism pos-int?)
```

Comment:
You might end up with a limit here.
```
The underlying http-client has a :pending-ops-limit configuration which, when reached, results in an exception with the message "Ops limit reached". As of this writing, aws-api does not provide access to the http-client's configuration. Programs that encounter "Ops limit reached" can avoid it by creating separate http-clients for each aws-client. You may wish to explicitly stop (com.cognitect.aws.api/stop) these aws-clients when the are not longer in use to conserve resources.
```
Of course I guess if you really wanted to you could use a seperate client instance for each client actor.